core/src/main/java/hudson/util/io/TarArchiver.java

```diff
@@ -59,7 +59,7 @@
     public void visitSymlink(File link, String target, String relativePath) throws IOException {
         TarArchiveEntry e = new TarArchiveEntry(relativePath, TarConstants.LF_SYMLINK);
         try {
-            int mode = IOUtils.mode(link);
```

Comment:
Oh, and if you insist, I'll be happy to duplicate the archive classes and add a checkbox "Yes, I want the old, broken behavior" somewhere in the GUI :-)